usr/share/rear/lib/layout-functions.sh

```diff
@@ -992,4 +992,180 @@ function apply_layout_mappings() {
     is_true $apply_layout_mappings_succeeded && return 0 || return 1
 }
 
+# Keeps track of last partition created for a specific disk
+# Associative array indexed by disk
+# e.g. last_partition_number[/dev/sda]=1
+declare -A last_partition_number=()
+
+# Keeps track of dummy partitions created and to be removed (due to parted limitation)
+# Associative array indexed by disk and containing a list of partition numbers
+# e.g. dummy_partitions_to_delete[/dev/sda]=" 1 2 4 5"
+declare -A dummy_partitions_to_delete=()
+
+# Keeps track of partitions to resize to original size
+# Associative array indexed by disk and containing a list of partition "number:end_in_bytes"
+# e.g. partitions_to_resize[/dev/sda]="3:2096127 6:8388607"
+declare -A partitions_to_resize=()
+
+# Keeps track of the label of a specific disk
+# Associative array indexed by disk
+# e.g. disk_label[/dev/sda]="gpt"
+declare -A disk_label=()
+
+create_disk_label() {
+    local disk="$1" label="$2"
+
+    LogPrint "Disk '$disk': creating '$label' partition table"
+    parted -s $disk mklabel $label >&2
+    my_udevsettle
+    disk_label[$disk]="$label"
+}
+
+create_disk_partition() {
+    local disk="$1" name="$2" number=$3 startB=$4 endB=$5
+
+    #
+    # FIXME? This code assumes that "parted" is capable of handling sizes in
+    # Bytes. parted supports partitions in Bytes since ages.
+    #
+
+    # The duplicated quoting "'$name'" is there because
+    # parted's internal parser needs single quotes for values with blanks.
+    # In particular a GPT partition name that can contain spaces
+    # like 'EFI System Partition' cf. https://github.com/rear/rear/issues/1563
+    # so that when calling parted on command line it must be done like
+    #    parted -s /dev/sdb unit MiB mkpart "'partition name'" 12 34
+    # where the outer quoting "..." is for bash so that
+    # the inner quoting '...' is preserved for parted's internal parser:
+    [ ${disk_label[$disk]} == "msdos" ] || name="'$name'"
+
+    [[ "${last_partition_number[$disk]}" ]] || last_partition_number[$disk]=0
+
+    if [[ $number -le ${last_partition_number[$disk]} ]] ; then
+        Error "Disk '$disk': trying to create partition number $number but last created partition was number ${last_partition_number[$disk]}"
+    fi
+
+    if [[ $(( $number - ${last_partition_number[$disk]} - 1 )) -eq 0 ]] ; then
+        LogPrint "Disk '$disk': creating partition number $number with name '$name'"
+        # FIXME: I <jsmeix@suse.de> think one cannot silently set the end of a partition to 100%
+        # if there is no partition end value, I think in this case "rear recover" should error out:
+        if [[ ! $endB ]] ; then
+            parted -s $disk mkpart "$name" "${startB}B" 100% >&2
```

Comment:
You do no longer need to redirect stdout to stderr
to aviod that unwanted stuff appears on the user's terminal because
since ReaR 2.3 both stderr and also stdout are redirected into the ReaR log file,
see the section "Version 2.3 (December 2017)" in
https://github.com/rear/rear.github.com/blob/master/documentation/release-notes-2-5.md
and see "What to do with stdin, stdout, and stderr" in
https://github.com/rear/rear/wiki/Coding-Style